packages/ember-htmlbars/tests/helpers/component_test.js

```diff
@@ -1,398 +0,0 @@
-  run(function() {
-    view.$('.foo-bar').trigger('click');
-  });
-  equal(actionTriggered, 1, 'action was triggered');
-});
-
-QUnit.test('component helper maintains expected logical parentView', function() {
-  owner.register('template:components/foo-bar', compile('yippie! {{yield}}'));
-  var componentInstance;
-  owner.register('component:foo-bar', Component.extend({
-    didInsertElement() {
-      componentInstance = this;
-    }
-  }));
-
-  view = EmberView.create({
-    [OWNER]: owner,
-    dynamicComponent: 'foo-bar',
-    template: compile('{{#component view.dynamicComponent}}arepas!{{/component}}')
-  });
-
-  runAppend(view);
-  equal(get(componentInstance, 'parentView'), view, 'component\'s parentView is the view invoking the helper');
-});
-
-QUnit.test('nested component helpers', function() {
-  owner.register('template:components/foo-bar', compile('yippie! {{attrs.location}} {{yield}}'));
-  owner.register('template:components/baz-qux', compile('yummy {{attrs.location}} {{yield}}'));
-  owner.register('template:components/corge-grault', compile('delicious {{attrs.location}} {{yield}}'));
-
-  view = EmberView.create({
-    [OWNER]: owner,
-    dynamicComponent1: 'foo-bar',
-    dynamicComponent2: 'baz-qux',
-    location: 'Caracas',
-    template: compile('{{#component view.dynamicComponent1 location=view.location}}{{#component view.dynamicComponent2 location=view.location}}arepas!{{/component}}{{/component}}')
-  });
-
-  runAppend(view);
-  equal(view.$().text(), 'yippie! Caracas yummy Caracas arepas!', 'components were looked up and rendered');
-
-  run(function() {
-    set(view, 'dynamicComponent1', 'corge-grault');
-    set(view, 'location', 'Loisaida');
-  });
-  equal(view.$().text(), 'delicious Loisaida yummy Loisaida arepas!', 'components were updated and re-rendered');
-});
-
-QUnit.test('component helper can be used with a quoted string (though you probably would not do this)', function() {
-  owner.register('template:components/foo-bar', compile('yippie! {{attrs.location}} {{yield}}'));
-
-  view = EmberView.create({
-    [OWNER]: owner,
-    location: 'Caracas',
-    template: compile('{{#component "foo-bar" location=view.location}}arepas!{{/component}}')
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), 'yippie! Caracas arepas!', 'component was looked up and rendered');
-});
-
-QUnit.test('component with unquoted param resolving to non-existent component', function() {
-  view = EmberView.create({
-    [OWNER]: owner,
-    dynamicComponent: 'does-not-exist',
-    location: 'Caracas',
-    template: compile('{{#component view.dynamicComponent location=view.location}}arepas!{{/component}}')
-  });
-
-  expectAssertion(function() {
-    runAppend(view);
-  }, /HTMLBars error: Could not find component named "does-not-exist"./, 'Expected missing component to generate an exception');
-});
-
-QUnit.test('component with unquoted param resolving to a component, then non-existent component', function() {
```

Comment:
moved to `@htmlbars component with static name argument for non-existent component`